File: src/test/java/no/utgdev/diy/ex7/MinComponent.java
package no.utgdev.diy.ex7;

import no.utgdev.diy.annotations.Component;
import no.utgdev.diy.annotations.Inject;
import no.utgdev.diy.annotations.PostConstruct;

@Component
public class MinComponent {
    @Inject
    private MinService service;

    private String serviceHei;

    public String hei() {
        return "Hei fra MinComponent og " + serviceHei;
    }

    @PostConstruct
    private void setup() {
        this.serviceHei = service.hei().toUpperCase();
    }
}

File: src/test/kotlin/no/utgdev/diy/ex7/MinComponent.kt
package no.utgdev.diy.ex7

//REMOVE import no.utgdev.diy.annotations.Component;
//REMOVE import no.utgdev.diy.annotations.Inject;
//REMOVE import no.utgdev.diy.annotations.PostConstruct;

//REMOVE @Component
class MinComponent {
    //REMOVE @Inject
    private val service: MinService? = null
    private var serviceHei: String? = null
    fun hei(): String {
        return "Hei fra MinComponent og $serviceHei"
    }

    //REMOVE @PostConstruct
    private fun setup() {
        serviceHei = service!!.hei().toUpperCase()
    }
}